### PR TITLE
Update diagnostic docs for SYSLIB0009

### PR DESCRIPTION
**Pull request by deeprobin**
## Summary

Follow-up for https://github.com/dotnet/runtime/pull/93171.
Change obsoletion documentation: Whole class marked as obsolete. 

See https://github.com/dotnet/runtime/issues/77459.

/cc @MihaZupan 

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/compatibility/9.0.md](https://github.com/dotnet/docs/blob/76a9165bdfd7d5c71aada1b2911b5038866861e5/docs/core/compatibility/9.0.md) | [Breaking changes in .NET 9](https://review.learn.microsoft.com/en-us/dotnet/core/compatibility/9.0?branch=pr-en-us-37452) |
| [docs/core/compatibility/core-libraries/9.0/obsolete-apis-with-custom-diagnostics.md](https://github.com/dotnet/docs/blob/76a9165bdfd7d5c71aada1b2911b5038866861e5/docs/core/compatibility/core-libraries/9.0/obsolete-apis-with-custom-diagnostics.md) | [API obsoletions with non-default diagnostic IDs (.NET 9)](https://review.learn.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/9.0/obsolete-apis-with-custom-diagnostics?branch=pr-en-us-37452) |
| [docs/fundamentals/syslib-diagnostics/obsoletions-overview.md](https://github.com/dotnet/docs/blob/76a9165bdfd7d5c71aada1b2911b5038866861e5/docs/fundamentals/syslib-diagnostics/obsoletions-overview.md) | [Obsolete features in .NET 5+](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/obsoletions-overview?branch=pr-en-us-37452) |
| [docs/fundamentals/syslib-diagnostics/syslib0009.md](https://github.com/dotnet/docs/blob/76a9165bdfd7d5c71aada1b2911b5038866861e5/docs/fundamentals/syslib-diagnostics/syslib0009.md) | [SYSLIB0009: AuthenticationManager is not supported](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib0009?branch=pr-en-us-37452) |

### CODE REVIEW

**Review by MihaZupan — APPROVED** [review]
Thank you

**Comment by gewarren** [comment]
Thank you. Since this is for .NET 9, we shouldn't merge it yet.

**Comment by gewarren** [comment]
We also need to:

- Update https://github.com/dotnet/docs/blob/main/docs/fundamentals/syslib-diagnostics/obsoletions-overview.md
- Create a new article like this but for .NET 9: https://github.com/dotnet/docs/blob/main/docs/core/compatibility/core-libraries/8.0/obsolete-apis-with-custom-diagnostics.md

File: docs/fundamentals/syslib-diagnostics/syslib0009.md
Comment (gewarren) [inline docs/fundamentals/syslib-diagnostics/syslib0009.md:7]:
```suggestion
# SYSLIB0009: AuthenticationManager is not supported
```

File: docs/fundamentals/syslib-diagnostics/syslib0009.md
Comment (gewarren) [inline docs/fundamentals/syslib-diagnostics/syslib0009.md:15]:
```suggestion
The methods in this class either no-op or throw a <xref:System.PlatformNotSupportedException> at run time.
```

File: docs/fundamentals/syslib-diagnostics/syslib0009.md
Comment (gewarren) [inline docs/fundamentals/syslib-diagnostics/syslib0009.md:14]:
```suggestion
In .NET 9 and later versions, the entire <xref:System.Net.AuthenticationManager> class is marked obsolete. Use of this class generates warning `SYSLIB0009` at compile time.
```

**Review by gewarren — CHANGES_REQUESTED** [review]

**Review by gewarren — APPROVED** [review]

**Comment by gewarren** [comment]
@wfurt It can be merged if we change the date that's causing a warning, but why publish .NET 9 documentation now when the first preview won't be available until February?

**Comment by wfurt** [comment]
so can complete runtime issue tracking it :)

File: docs/core/compatibility/core-libraries/9.0/obsolete-apis-with-custom-diagnostics.md
Comment (gewarren) [inline docs/core/compatibility/core-libraries/9.0/obsolete-apis-with-custom-diagnostics.md:5]:
```suggestion
ms.date: 04/05/2024
```

File: docs/core/compatibility/9.0.md
Comment (gewarren) [inline docs/core/compatibility/9.0.md:65]:
```suggestion
- [What's new in .NET 9](../whats-new/dotnet-9/overview.md)
```

**Pull request merged**
